controllers/virtualmachinepublishrequest/virtualmachinepublishrequest_controller.go

```diff
@@ -944,19 +998,45 @@ func getPublishRequestActID(vmPub *vmopv1.VirtualMachinePublishRequest) string {
 
 // parseItemIDFromTaskResult returns the content library item UUID from the task result.
 func parseItemIDFromTaskResult(result vimtypes.AnyType) (string, error) {
-	clItem, ok := result.(vimtypes.ManagedObjectReference)
+	item, ok := result.(vimtypes.ManagedObjectReference)
 	if !ok {
 		return "", fmt.Errorf("failed to cast task result to ManagedObjectReference")
 	}
 
-	if clItem.Type != "ContentLibraryItem" {
-		return "", fmt.Errorf("expect task result type to be ContentLibraryItem, get %s instead", clItem.Type)
+	switch item.Type {
+	case "ContentLibraryItem":
+		if !strings.HasPrefix(item.Value, clItemPrefix) {
+			return "", fmt.Errorf("content library item UUID is invalid, value: %s", item.Value)
+		}
+		return item.Value[len(clItemPrefix):], nil
+	case "VirtualMachine":
+		return item.Value, nil
+	default:
+		return "", fmt.Errorf("expect task result type to be ContentLibraryItem or VirtualMachine, got %s instead", item.Type)
+	}
+}
+
+func (r *Reconciler) getTargetLibrary(ctx *pkgctx.VirtualMachinePublishRequestContext) error {
+	targetLocationName := ctx.VMPublishRequest.Spec.Target.Location.Name
+	objKey := client.ObjectKey{Name: targetLocationName, Namespace: ctx.VMPublishRequest.Namespace}
+
+	contentLibrary := &imgregv1a1.ContentLibrary{}
```

Comment:
It makes sense to call get r.Get(ctx, objKey, contentLibraryV1A2) every reconcile loop since we need ContentLibraryType throughout different stages of the process. JW if we should keep r.Get(ctx, objKey, contentLibraryV1A1) as it is on where we are reading it.